oxo.py

```diff
@@ -1,26 +1,50 @@
 class OxoBoard:
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.board = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        return self.board[x][y]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.board[x][y] == 0:
+            self.board[x][y] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        for x in range(len(self.board)):
+            for y in range(len(self.board[x])):
+                if self.board[x][y] == 0:
+                    return False
+        return True
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+        if self.board[0][0] == self.board[0][1] == self.board[0][2]:
```

Comment:
You also need to check if the cells in the line are not all zero, e.g.:
```
if self.board[0][0] == self.board[0][1] == self.board[0][2] != 0:
```
Otherwise if an empty line is detected before a winning line, the function will incorrectly return 0. This is what is causing some of the test cases to fail.